main_test.go

```diff
@@ -206,3 +206,33 @@ func TestMPL(t *testing.T) {
 		t.Fatalf("%v\n%s", err, out)
 	}
 }
+
+func TestExcludePath(t *testing.T) {
+	if os.Getenv("RUNME") != "" {
+		main()
+		return
+	}
+
+	tmp := tempDir(t)
+	t.Logf("tmp dir: %s", tmp)
+
+	origin1 := filepath.Join(tmp, "file.c")
+	origin2 := filepath.Join(tmp, "file.h")
+	expected1 := "testdata/initial/file.c"
+	expected2 := "testdata/expected/file.h"
+
+	run(t, "cp", "testdata/initial/file.c", origin1)
+	run(t, "cp", "testdata/initial/file.h", origin2)
+
+	cmd := exec.Command(os.Args[0],
+		"-test.run=TestExcludePath",
+		"-y", "2018", "-e", origin1, origin1, origin2,
```

Comment:
Two questions: 1) this is excluding a file. Isn't the more likely use case to exclude an entire dir tree? If so, could you add a test for that case? 2) Is the two appearances of origin1 a mistake or because you want to verify the logic handles duplicate paths?

Reply:
1. In this implementation, I expect that only to accept filename. for example incurrent implementation we get `/foo/bar`, `/foo/bar/baz.txt` and configured `/foo/bar` as exclude path then `/foo/bar/baz.txt` won't be ignored.
2. It also includes dup path test.

Reply:
I think it'd be beneficial to support directories: the main use case (and ours!) is dev teams removing test directories from the path. The directory has a number of files that change over time, so it's not feasible to update the command every time a new file is added.

Reply:
I agree - I think dirs is likely to be a far more valuable use case than enumerating individual files. If you do a prefix match then you get both, dirs and files, with a one line change.